### ISSUE TITLE
Strange behavior at higher DPI or Resolution

### ISSUE THREAD

**Issue by TiKevin83** [google/guetzli#250]
Guetzli works really well for compressing some of the images in my library for the web.  I've managed to find a compression sweet spot targeting a butteraugli distance of 1.15 for many of the smaller JPEGs I'm working with (around 200x300), where guetzli+JpegTran performs equal to or better than mozjpeg consistently past 2.5 bits/pixel.

However, the sweet spot seems to disappear when trying to compress larger resolution originals of the same image (around 450x650).  Guetzli+JpegTran can produce an image at butteraugli 1.15 with roughly 2.25 bits/pixel at this resolution in my image library (roughly 10% fewer bits/pixel than the smaller size) while a similar quality mozjpeg encode is available with as little as 1.4-1.5 bits/pixel.

I could easily be missing something but it seems counter-intuitive that guetzli would be less efficient with higher DPI when it's explicitly tuned for high image quality.

**Comment by jyrkialakuijala** [comment]
Butteraugli has been optimized for the human vision at a viewing distance of 1000 pixels. When the viewing distance is more, some other approximations can be better. Also, it can be that your originals are in YUV420 and the higher resolution originals have the YUV420 compatible color characteristics. If that is the case, the YUV420 model in mozjpeg may be a better match than Guetzli's eagerness to apply YUV444.

Please send sample images for more informed speculation :-)

**Comment by jyrkialakuijala** [comment]
Would "Non-competitive density" be a better word for title than "Strange" ?

**Comment by jyrkialakuijala** [comment]
Without seeing the images, I suspect that these images have a certain level of noise that is present in the higher resolution images, but not in the lower resolution subsampled images. The noise is replicated by guetzli -- (as guetzli's philosophy is to be true to the original whatever it is), but cleaned by mozjpeg. Not having to replicate noise makes mozjpeg perform better.

**Comment by TiKevin83** [comment]
Yes that would be more accurate, thank you for your feedback!  I'll see what I can do about images.  The example here involves close-up (within a foot viewing distance) detail of a hand and a jar of coins with lots of ridge detail.  The ridge detail appears to disappear in visible blocks as butteraugli distance goes from 1 to 2 instead of uniformly throughout the picture, and some blocks also tend to align vertically which is more readily visible than mozjpeg's degredation.

**Comment by TiKevin83** [comment]
Here's the set of images that have been giving me the most trouble at high DPI.  Mozjpeg 3.1 at Q85, Guetzli at Q86+JpegTran, and original:

![mozjpeg85](https://user-images.githubusercontent.com/38826675/41417944-b656c724-6fbc-11e8-8749-97a5aed0e526.jpg)
![guetzliq86 jpegtran](https://user-images.githubusercontent.com/38826675/41417946-b6681916-6fbc-11e8-86a0-e9df17c36b8a.jpg)
![cd-special-card670w](https://user-images.githubusercontent.com/38826675/41417958-bd075886-6fbc-11e8-9fd2-f29685780d57.png)

On the Guetzli image, ridge detail on knuckles and in the crease between the thumb+pointer finger is preserved compared to mozjpeg, but much more visible artifacts appear in the green between the 3rd stack of coins and the coin jar, and the shape of some of the background halos is distorted due to blocking.  There's also blocking in the coin jar that creates noticeable vertical lines through the image, and the same at the base of the pointer finger and the middle finger.  From what I can see, Guetzli+JpegTran begins to be visually competitive around Q92-93, but with 40% more bits/pixel.

**Comment by TiKevin83** [comment]
I've made some progress troubleshooting the issue.  I think in my case, many of the stock images we use were at some point 4:2:0 before being converted to pngs, so they're only efficiently compressed at 4:4:4 after being downsampled past 2x.  When I switch force_420 to true in processor.h, the guetzli+jpegtran output at Q86 is much closer to what I would expect.
![cd-special-card670w-force420](https://user-images.githubusercontent.com/38826675/41547994-adda7ff6-72ef-11e8-94fc-a6121b0b883e.jpg)
However, I can see why force_420 is normally avoided, as it causes a large unavoidable butteraugli distance on 4:4:4 input.

I am still seeing though that 4:4:4 images ran through my [auto-quality search](https://github.com/TiKevin83/guetzli) to minimize filesize/butteraugli distance tradeoffs pretty consistently have lower BA distance with smaller and smaller subsampled input.  I'm wondering if this is a consequence of targeting a 1000 pixel viewing distance or if something else is going on.

**Comment by TiKevin83** [comment]
New encode targeting BA distance 1.632 + JpegTran for equivalent filesize as the previous Q86 (BA target 1.8057)+JpegTran:
![cd-special-card670q1 632](https://user-images.githubusercontent.com/38826675/42392432-d07dd28a-8120-11e8-8dc1-5a112c26bec1.jpg)

New Guetzli appears favorable mainly in the ringing around the plant, but it also appears to have introduced a new artifact pattern in the white lines above the tallest stack of coins.  Both new and old have blockiness issues, primarily with how they degrade the ridge detail in the hand in general: where old has stripes of blocks that lose ridge detail, new Guetzli tends to have stripes of blocks that preserve some of the detail.
